tempodb/tempodb.go

```diff
@@ -242,6 +242,15 @@ func (rw *readerWriter) Find(ctx context.Context, tenantID string, id common.ID,
 			copiedBlocklist = append(copiedBlocklist, b)
 		}
 	}
+
+	// if block is compacted within lookback period, include it in search
+	lookback := time.Now().Add(-(2 * rw.cfg.BlocklistPoll))
+	compactedBlocklist := rw.compactedBlockLists[tenantID]
+	for _, c := range compactedBlocklist {
+		if c.CompactedTime.After(lookback) {
```

Comment:
this also needs to take into account the block id range.  please do something similar to the `includeBlock` method and test (or extend the includeBlock method if possible)